File: app/src/main/java/com/shveed/cookmegood/data/RuntimeStorage.java
package com.shveed.cookmegood.data;

import java.util.ArrayList;
import java.util.List;

public class RuntimeStorage {

    private static RuntimeStorage instance;

    public List<String> categories = new ArrayList<>();

    public static RuntimeStorage newInstance(){
        if(instance == null){
            instance = new RuntimeStorage();
        }
        return instance;
    }


}

File: app/src/main/java/com/shveed/cookmegood/data/RuntimeStorage.kt
package com.shveed.cookmegood.data

class RuntimeStorage {
    var categories = arrayListOf<String>()

    companion object {
        private var instance: RuntimeStorage? = null
        fun newInstance(): RuntimeStorage? {
            if (instance == null) {
                instance = RuntimeStorage()
            }
            return instance
        }
    }
}